experimental: Fix graph_transformers llms #21482

liugz18 opened the pull request:
Fix AttributeError on calling LLMGraphTransformer.convert_to_graph_documents #21482

 since raw_schema is always a str

@baskaryan

liugz18 committed:
Fix AttributeError on calling LLMGraphTransformer.convert_to_graph_documents #21482

Fix AttributeError on calling LLMGraphTransformer.convert_to_graph_documents #21482
since raw_schema is always a str

baskaryan commented:
@tomasonjo

tomasonjo commented:
I don't think it's always a string as this is how it's defined:

```
self.chain = prompt | llm
```

I've also tested this approach with Ollama.
Perhaps there are some inconsistencies between different LLM providers...
Can you change the code so that it checks if raw_schema isn't a string it extracts content attribute, otherwise it doesn't do anything?

liugz18 committed:
Merge branch 'master' into master

liugz18 committed:
Update llm.py

change the code so that it checks if raw_schema isn't a string it extracts content attribute, otherwise it doesn't do anything

liugz18 committed:
Update llm.py

change the code so that it checks if raw_schema isn't a string it extracts content attribute, otherwise it doesn't do anything

tomasonjo commented:
Thanks @liugz18 ... cc @baskaryan
